Lib/test/test_shlex.py

```diff
@@ -339,6 +343,22 @@ def testQuote(self):
             self.assertEqual(shlex.quote("test%s'name'" % u),
                              "'test%s'\"'\"'name'\"'\"''" % u)
 
+    def testQuoteBytes(self):
+        """Test quoting of byte objects"""
+        # Copied from testQuote
+        safeunquoted = string.ascii_letters + string.digits + '@%_-+=:,./'
+        unsafe = '"`$\\!'
+
+        self.assertEqual(shlex.quote(b''), b"''")
```

Comment:
Would be good to test some bytestrings with non-ASCII characters, to make sure that they're handled correctly.

Reply:
I added the missing unicode characters from the `testQuote` test, and changes the encoding to utf-8 to account for the changes. Added in bfb76be9.

Reply:
Please add *all* characters that are supported on POSIX (not just three of them):

```py
unicode_chars = ('ßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿ'
 				 'ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞ')
```